services/tracing/tracing/application.go

```diff
@@ -0,0 +1,44 @@
+package tracing
+
+import "sensorbucket.nl/sensorbucket/pkg/pipeline"
+
+type StepStore interface {
+	Insert(Step) error
+}
+
+func New(stepStore StepStore) *Service {
+	return &Service{
+		stepStore: stepStore,
+	}
+}
+
+type Service struct {
+	stepStore StepStore
+}
+
+func (s *Service) HandlePipelineMessage(pipelineMessage pipeline.Message) error {
+	return s.addStep(Step{
+		TracingID: pipelineMessage.ID,
+		StepIndex: pipelineMessage.StepIndex,
+
+		// We have to add 1 to the stepindex to get the actual steps remaining
+		StepsRemaining: int64(len(pipelineMessage.PipelineSteps) - (int(pipelineMessage.StepIndex + 1))),
```

Comment:
Just an idea, what if we just make this "TotalSteps"? A user could then choose to calculate stepsremaining if required.
Are we using `StepsRemaining` explicitly somewhere?

Reply:
maybe for in the future. In the end I don't think it matters. It might be useful for debugging to view how many steps were considered to be remaining at each step. Then again, both can be calculated with either choice